pypesto/hierarchical/optimal_scaling/problem.py

```diff
@@ -429,29 +496,76 @@ def optimal_scaling_inner_parameters_from_measurement_df(
         InnerParameterType.OPTIMAL_SCALING
     ].values()
 
-    for par_type, par_estimate in zip(par_types, estimate):
-        for _, row in df.iterrows():
-            if row[MEASUREMENT_TYPE] == ORDINAL:
-                par_id = f'{par_type}_{row[OBSERVABLE_ID]}_{row[MEASUREMENT_GROUP]}_{row[MEASUREMENT_CATEGORY]}'
-
-                # Create only one set of bound parameters per category of a
-                # group.
-                if par_id not in [
-                    inner_par.inner_parameter_id
-                    for inner_par in inner_parameters
-                ]:
-                    inner_parameters.append(
-                        OptimalScalingParameter(
-                            inner_parameter_id=par_id,
-                            inner_parameter_type=InnerParameterType.OPTIMAL_SCALING,
-                            scale=LIN,
-                            lb=lb,
-                            ub=ub,
-                            category=row[MEASUREMENT_CATEGORY],
-                            group=row[MEASUREMENT_GROUP],
-                            estimate=par_estimate,
+    for observable_id in observable_ids:
+        group = observable_ids.index(observable_id) + 1
+
+        observable_df = df[df[OBSERVABLE_ID] == observable_id]
+
+        if all(observable_df[MEASUREMENT_TYPE] == ORDINAL):
+            # Add optimal scaling parameters for ordinal measurements.
+            for par_type, par_estimate in zip(par_types, estimate):
+                for _, row in observable_df.iterrows():
+                    par_id = f'{par_type}_{observable_id}_{row[MEASUREMENT_TYPE]}_{int(row[MEASUREMENT_CATEGORY])}'
+
+                    # Create only one set of bound parameters per category of a group.
+                    if par_id not in [
+                        inner_par.inner_parameter_id
+                        for inner_par in inner_parameters
+                    ]:
+                        inner_parameters.append(
+                            OptimalScalingParameter(
+                                inner_parameter_id=par_id,
+                                inner_parameter_type=InnerParameterType.OPTIMAL_SCALING,
+                                scale=LIN,
+                                lb=lb,
+                                ub=ub,
+                                observable_id=observable_id,
+                                category=int(row[MEASUREMENT_CATEGORY]),
+                                group=group,
+                                estimate=par_estimate,
+                            )
                         )
+        elif any(observable_df[MEASUREMENT_TYPE].isin(CENSORING_TYPES)):
+            # Get df with only censored measurements.
+            censored_df = observable_df.loc[
+                df[MEASUREMENT_TYPE].isin(CENSORING_TYPES)
```

Comment:
```suggestion
                observable_df[MEASUREMENT_TYPE].isin(CENSORING_TYPES)
```
(not sure)

Reply:
By chance, they turn out to be the same, but it should be the thing you suggested :D Could've been a nasty bug to debug.